File: 백준/Silver/1929. 소수 구하기/소수 구하기.cc
#include <iostream>
#include <vector>
using namespace std;

vector<int> findPrimes(int M, int N) {
    // 에라토스테네스의 체 배열 초기화
    vector<bool> isPrime(N + 1, true);
    isPrime[0] = isPrime[1] = false;
    
    // 에라토스테네스의 체 알고리즘
    for (int i = 2; i * i <= N; i++) {
        if (isPrime[i]) {
            // i의 배수들을 모두 false로 표시
            for (int j = i * i; j <= N; j += i) {
                isPrime[j] = false;
            }
        }
    }
    
    // M이상 N이하의 소수만 결과 벡터에 담기
    vector<int> primes;
    for (int i = M; i <= N; i++) {
        if (isPrime[i]) {
            primes.push_back(i);
        }
    }
    
    return primes;
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    int M, N;
    cin >> M >> N;
    
    vector<int> primes = findPrimes(M, N);
    
    // 결과 출력
    for (int prime : primes) {
        cout << prime << '\n';
    }
    
    return 0;
}

File: 백준/Silver/1929. 소수 구하기/소수 구하기.py
def find_primes(M: int, N: int) -> list:
    # 에라토스테네스의 체 배열 초기화
    is_prime = [True] * (N + 1)
    is_prime[0] = is_prime[1] = False
    
    # 에라토스테네스의 체 알고리즘
    for i in range(2, int(N ** 0.5) + 1):
        if is_prime[i]:
            # i의 배수들을 모두 False로 표시
            for j in range(i * i, N + 1, i):
                is_prime[j] = False
    
    # M이상 N이하의 소수 리스트 생성
    primes = [num for num in range(M, N + 1) if is_prime[num]]
    return primes

# 입력 받기
M, N = map(int, input().split())

# 소수 찾기 및 출력
primes = find_primes(M, N)
for prime in primes:
    print(prime)